pep-0669.rst

```diff
@@ -0,0 +1,318 @@
+PEP: 669
+Title: Low Impact Instrumentation and Monitoring for CPython
+Author: Mark Shannon <mark@hotpy.org>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 18-Aug-2021
+Post-History: 13-Sep-2021
+
+
+Abstract
+========
+
+Using a profiler or debugger in CPython can have a severe impact on
+performance. Slowdowns by an order of magnitude are not uncommon.
+It does not have this bad.
+
+This PEP proposes an API for instrumentation and monitoring of Python
+programs running on CPython that will enable the insertion of instrumentation
+and monitoring at low cost.
+
+Using the new API, code run under a debugger on 3.11 should easily outperform
+code run without a debugger on 3.10.
+
+Profiling will still slow down execution, but by much less than in 3.10.
+
+Motivation
+==========
+
+Developers should not have to pay an unreasonable cost to use debuggers,
+profilers and other similar tools.
+
+C++ and Java developers expect to be able to run a program at full speed
+(or very close to it) under a debugger.
+Python developers should expect that too.
```

Comment:
For those who think that `pdb` is the peak of debugging, and don't suffer from any performance issues with that, is there any motivation to support the change?

I'm obviously in agreement that a better approach would be great, just concerned that there may be push back from people who don't see the value when a few lines here about the current state of things (maybe the frequent checks for tracing?) would probably motivate things.